src/main/java/it/gov/pagopa/cgn/portal/converter/backoffice/approved/BackofficeApprovedAgreementConverter.java

```diff
@@ -33,6 +50,7 @@ protected Function<ApprovedAgreement, ApprovedAgreementEntity> toEntityFunction(
         dto.setAgreementStartDate(entity.getStartDate());
         dto.setPublishedDiscounts(entity.getPublishedDiscounts());
         dto.setTestPending(entity.getTestPending());
+        dto.setEntityType(getEntityTypeFromEntityTypeEnum(entity.getProfile().getEntityType()));
```

Comment:
why taking the entity type from profile if we should have saved it on agreement?

Reply:
changed ApprovedAgreementEntity in order to retrieve also entity type from AgreementEntity